### PR TITLE
Improve TypeHinting

### PR DESCRIPTION
**Pull request by LKaemmerling**
Signed-off-by: Lukas Kämmerling <lukas.kaemmerling@hetzner-cloud.de>

This MR basically brings the same improvements @rdohms did within https://github.com/PromPHP/prometheus_client_php/pull/17 to the push_gateway client.

### CODE REVIEW

File: src/PrometheusPushGateway/PushGateway.php [hunk 1]
```diff
@@ -26,9 +29,9 @@ class PushGateway
     /**
      * PushGateway constructor.
      * @param string $address (http|https)://host:port of the push gateway
-     * @param ClientInterface $client
+     * @param ClientInterface|null $client
      */
-    public function __construct($address, ClientInterface $client = null)
+    public function __construct(string $address, ClientInterface $client = null)
```
Comment (rdohms) [inline src/PrometheusPushGateway/PushGateway.php]: was thinking, should this be `?ClientInterface` ?

File: src/PrometheusPushGateway/PushGateway.php [hunk 2]
```diff
@@ -39,50 +42,50 @@ public function __construct($address, ClientInterface $client = null)
      * Uses HTTP PUT.
      * @param CollectorRegistry $collectorRegistry
      * @param string $job
-     * @param array $groupingKey
+     * @param array<string> $groupingKey
```
Comment (rdohms) [inline src/PrometheusPushGateway/PushGateway.php:45]: we ended up with `string[]` and `array<string>` i think I'll try to update the other project to use the later syntax as well.

**Review by rdohms — APPROVED** [review]
Looks great! 🎉

File: src/PrometheusPushGateway/PushGateway.php (hunk 1 above)
Comment (LKaemmerling) [reply src/PrometheusPushGateway/PushGateway.php]: yep you are right, will change it.

File: src/PrometheusPushGateway/PushGateway.php (hunk 2 above)
Comment (LKaemmerling) [reply src/PrometheusPushGateway/PushGateway.php:45]: Actually, this is a quite "special" case. It needs to be an assoc string array, therefore i decided to use this notation. Within the PHP client, it is an "array" with strings as values.

File: src/PrometheusPushGateway/PushGateway.php (hunk 1 above)
Comment (LKaemmerling) [reply src/PrometheusPushGateway/PushGateway.php]: @rdohms i changed it.

File: src/PrometheusPushGateway/PushGateway.php (hunk 2 above)
Comment (LKaemmerling) [reply src/PrometheusPushGateway/PushGateway.php:45]: @rdohms can i resolve this or should i change it to another notation?

Comment (rdohms) [reply src/PrometheusPushGateway/PushGateway.php:45]: LGTM

**Pull request merged**